### PR TITLE
Introduce a TokenList struct

### PR DESCRIPTION
**Pull request by fleupold**
This will allow us to easily configure a whitelist of tokens for which we are happy to market make using internal buffers. I'm thinking about using https://tokens.coingecko.com/uniswap/all.json

### Test Plan
Unit tests.

### CODE REVIEW

**Review by vkgnosis — APPROVED** [review]

**Comment by bh2smith** [comment]
One concern I have with the tokenlist you suggested is that it contains NAWA (a token that was recently rug pulled).

File: shared/src/macros.rs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#[macro_export]
+macro_rules! addr {
+    ($val:literal) => {
+        ::ethcontract::H160(::hex_literal::hex!($val))
+    };
```
Comment (bh2smith) [inline shared/src/macros.rs:5]: `Fancy!(); MacroMan!();`

**Review by bh2smith — APPROVED** [review]
Very nice `TokenList` struct and implemented functions. I can see this coming in handy for other projects.

File: shared/src/macros.rs (hunk 1 above)
Comment (fleupold) [reply shared/src/macros.rs:5]: this was just move, so not my 🧑‍🎨

**Comment by fleupold** [comment]
> One concern I have with the tokenlist you suggested is that it contains NAWA (a token that was recently rug pulled). It is quite a vast list and maybe we want to tone the default down a bit.

The list can be configured via config arguments, so we can use whatever condensed list we like.

**Pull request merged**